In [2]:
import pandas as pd
path = "shopping_trends.csv"
df = pd.read_csv(path)
df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,Cash,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,PayPal,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Credit Card,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,PayPal,Express,No,No,24,Venmo,Weekly


Mencari Rata rata (kolom 18)

In [8]:
df_noid = df.iloc[:,16:]
df_noid

,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,14,Venmo,Fortnightly
1,2,Cash,Fortnightly
2,23,Credit Card,Weekly
3,49,PayPal,Weekly
4,31,PayPal,Annually
...,...,...,...
3895,32,Venmo,Weekly
3896,41,Bank Transfer,Bi-Weekly
3897,24,Venmo,Quarterly
3898,24,Venmo,Weekly


In [9]:
df_noid[['Previous Purchases']].mean()

Previous Purchases    25.351538
dtype: float64

std

In [ ]:
df_noid[['Previous Purchases']].std()

Previous Purchases    14.447125
dtype: float64

min

In [11]:
df_noid[['Previous Purchases']].min()

Previous Purchases    1
dtype: int64

max

In [12]:
df_noid[['Previous Purchases']].max()

Previous Purchases    50
dtype: int64

Q1, Q2, Q3

In [16]:
df_noid.describe()

,Previous Purchases
count,3900.000000
mean,25.351538
std,14.447125
min,1.000000
25%,13.000000
50%,25.000000
75%,38.000000
max,50.000000


Analisa korelasi Age vs Purchase Amount (USD) dan Purchase Amount (USD) vs Review
RatinG

In [18]:
df_noid = df.iloc[:,1:]
df_noid

,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,Cash,2-Day Shipping,No,No,32,Venmo,Weekly
3896,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,PayPal,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Credit Card,Standard,No,No,24,Venmo,Quarterly
3898,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,PayPal,Express,No,No,24,Venmo,Weekly


In [19]:
df_noid[['Age','Purchase Amount (USD)']].corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


Dari data di atas, dapat dipahami bahwa:
*  Age vs Purchase Amount (USD) (-0.010424), berarti semakin naik usia maka semakin sedikit pengeluaran 

In [20]:
df_noid[['Purchase Amount (USD)','Review Rating']].corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


Dari data di atas, dapat di pahami bahwa:
* Purchase Amount (USD) vs Review Rating (+0.030776), berarti semakin banyak jumlah pembelian maka rating yang di dapat semakin banyak